lib/puppetserver/ca/action/list.rb

```diff
@@ -79,15 +89,52 @@ def run(input)
           requested, signed, revoked = separate_certs(all_certs)
           missing = certnames - all_certs.map { |cert| cert['name'] }
 
-          (all || certnames.any?) \
+          if output_format == 'json'
+            (all || certnames.any?)\
+            ? output_certs_json_format(all, requested, signed, revoked, missing)
+            : output_certs_json_format(all, requested)
+          else
+            (all || certnames.any?)\
```

Comment:
Nice.

Now instead of duplicating the requested vs all logic in a formatting conditional I think we should wrap the call to `output_certs_json_format` and `output_certs_by_state` with one method (maybe `output_certs_by_state` and then rename the current method to `output_certs_text_format`) that calls the correct formatter based a flag you pass. Then the logic of which certs to display is totally separate from the logic about how to format them

Reply:
Your wish is my command 🙇🏼‍♂️